iquip/apps/monitor.py

```diff
@@ -108,114 +163,148 @@ def __init__(
             ttlWidgetLayout.addWidget(ttlWidget, row, column)
-        self.button = QPushButton("Not Overriding?", self)
-        self.button.setCheckable(True)
+        overrideButtonBox = QGroupBox("Override", self)
+        self.overrideOnButton = QPushButton("ON", self)
+        self.overrideOffButton = QPushButton("OFF", self)
         # layout
+        overrideButtonLayout = QHBoxLayout(overrideButtonBox)
+        overrideButtonLayout.addWidget(self.overrideOnButton)
+        overrideButtonLayout.addWidget(self.overrideOffButton)
         layout = QVBoxLayout(self)
         layout.addLayout(ttlWidgetLayout)
-        layout.addWidget(self.button)
+        layout.addStretch()
+        layout.addWidget(overrideButtonBox)
         # signal connection
-        self.button.clicked.connect(self.overrideChanged)
-        self.overrideChanged.connect(self._setButtonText)
+        self.overrideOnButton.clicked.connect(lambda: self.overrideChangeRequested.emit(True))
+        self.overrideOffButton.clicked.connect(lambda: self.overrideChangeRequested.emit(False))
 
-    @pyqtSlot(bool)
-    def _setButtonText(self, override: bool):
-        """Sets the button text.
+
+class _TTLStatusThread(QThread):
+    """QThread for fetching the TTL status from the proxy server.
+    
+    Signals:
+        fetched(modifications): The modifications of TTL status is fetched.
+          The "modifications" is a dictionary with three keys; "probe", "level", and "override".
+          Its value is a dictionary whose key is a TTL name and value is the modified value.
+
+    Attributes:
+        url: Web socket url.
+        devices: List of TTL names.
+    """
+
+    fetched = pyqtSignal(dict)
+
+    def __init__(self, ip: str, port: int, devices: List[str], parent: Optional[QObject] = None):
+        """Extended.
         
         Args:
-            override: Whether the button is now checked or not.
+            ip: proxy server IP address.
+            port: proxy server PORT number.
+            devices: See attribute section.
         """
-        if override:
-            self.button.setText("Overriding")
-        else:
-            self.button.setText("Not Overriding")
+        super().__init__(parent=parent)
+        self.url = f"ws://{ip}:{port}/ttl/status/modification/"
+        self.devices = devices
+
+    def run(self):
+        """Overridden.
+        
+        Fetches the modifications of TTL status from the proxy server.
+
+        Whenever fetched, the fetched signal is emitted.
+        """
+        try:
+            with connect(self.url) as websocket:
+                websocket.send(json.dumps(self.devices))
+                for response in websocket:
+                    status = json.loads(response)
+                    self.fetched.emit(status)
+        except WebSocketException:
+            logger.exception("Failed to fetch the modifications of TTL status.")
 
 
 class _TTLOverrideThread(QThread):
```

Comment:
Maybe we can design a simple HTTP POST requester for convenience later (not now).

Reply:
I was also feeling its need. I think we should make a subclass of `QThread` supporting a general HTTP communication.